CRM/Utils/Mail.php

```diff
@@ -205,7 +205,12 @@ public static function send(array &$params): bool {
 
     if (is_object($mailer)) {
       try {
-        $result = $mailer->send($to, $headers, $message ?? '');
+        $originalValues = [
+          'html' => $params['html'] ?? NULL,
+          'text' => $params['text'] ?? NULL,
+          'attachments' => $params['attachments'] ?? [],
```

Comment:
@eileenmcnaughton Please can we add in 'Bcc' => $headers['Bcc'] ?? NULL or similar here because of this crazyness: https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/Mail.php#L202 which makes it really annoying to get the Bcc recipients back into the headers if they are required. You have to parse the headers, extract the email addresses and compare against the $to array! The missing ones then have to be added back in as Bcc.
At least that's the case for eg. Amazon SES via API.

Reply:
Note, we'd need to store the value before unsetting it as well.

Reply:
Oh, it *might* make sense to just add in Cc array keys here as well for completeness though we don't necessarily need it because we can find it in the headers. Would make it easier for anyone wanting to work with this in the future though.

Reply:
@mattwire I feel like if we add in Bcc we should add in to, cc & bcc (maybe from too) - but then the format becomes the question - from context it looks like we have already formatted the emails so an array like this probably makes sense (& invites the addition of 'from' too) - ie it's an array of strings where name & email are already formatted into a single string, if both exist

```
'to' => [
  'matt@wire.com',
  "Matt Wire <matt@anotherwire.com">,
],
'cc' =>  ['matt_the_cc@wire.com', "Matt Wire <matt_the_cc@anotherwire.com">],
'bcc' =>  ['matt_the_bcc@wire.com', "Matt Wire <matt_the_bcc@anotherwire.com">],
```

Reply:
Hmm - I guess what you are saying is that ^^ would be no different to what is currently in 'headers'

Reply:
@mattwire I've added in 'bcc' - I'm on the fence about `cc` cos I feel like that would be `cc`, `to` and `from` once we get into the completeness question - which might make sense but I'm not sure.

Also, my read of the code is that Bcc would currently be a single email address so converting to an array through casting works - is that your take?

Reply:
Oh dang - it comes in as 

`"Adams, Clint" <adams.j.clint@example.org>,"admin@example.com" <admin@example.com>`

I kinda hate not converting it to an array - but I don't think I can safely explode it

Reply:
Here is my workaround! https://lab.civicrm.org/extensions/ses/-/commit/64344002c28daf4f0f54abcf186c3b8f18bfb5e3